src/module/TableOperationMenu.js

```diff
@@ -159,9 +160,64 @@ export class TableOperationMenu {
     optionsMerge() {
         if (this.options?.replaceItems) {
             this.menuItems = this.options?.items ?? {};
-        } else {
+        } else if (!this.options?.modifyItems) {
             this.menuItems = Object.assign({}, MENU_ITEMS_DEFAULT, this.options?.items ?? {});
+        } // The menuItems for the other condition is managed through the optionsModify function.
+    }
+    /**
+     * Rewrite the whole functionality in case of 'setBorderColor' and 'setBackgroundColor' buttons.
+     * @param {string} item The type of the button
+     * @param {object} itemNewOptions Contains the user-defined attributes get used to override the buttons' properties
+     */
+    overrideButton(item, itemNewOptions) {
+        MENU_ITEMS_DEFAULT[item] = {
+            text() {
+                const subTitle = document.createElement('span');
+                subTitle.innerText = itemNewOptions.text;
+                const tableModule = this.quill.getModule(moduleName.table);
+                const input = document.createElement('input');
+                input.type = 'color';
+                input.addEventListener('click', (e) => {
+                    e.stopPropagation();
+                });
+                const tempCells = tableModule.tableSelection.selectedTds;
+                input.addEventListener('input', () => {
+                    switch (item) {
+                        case 'setBorderColor':
+                            tableModule.setStyle({ borderColor: input.value }, tempCells);
+                            break;
+                        case 'setBackgroundColor':
+                            tableModule.setStyle({ backgroundColor: input.value }, tempCells);
+                            break;
+                    }
+                });
+                input.style.marginLeft = 'auto';
+                return [subTitle, input];
+            },
+        }
+    }
+    /**
+     * Override the attributes of the context menu items if they exist;
+     * otherwise, define a new one for them.
+     */
+    optionsModify() {
+        if (!this.options?.modifyItems) return;
+        for (const [item, itemNewOptions] of Object.entries(this.options?.items)) {
+            if (!MENU_ITEMS_DEFAULT.hasOwnProperty(item)) continue;
+            switch (item) {
+                case 'setBorderColor':
+                case 'setBackgroundColor':
+                    this.overrideButton(item, itemNewOptions);
+                    break;
+                default:
+                    const itemDefaultOptions = MENU_ITEMS_DEFAULT[item];
+                    Object.keys(itemNewOptions).forEach(option => {
```

Comment:
Replacing 'forEach' with 'for ... of' would be better

Reply:
I chose the forEach method since the itemNewOptions are far unlikely to contain a large number of attributes, and the performance impact is almost negligible. btw, I'll do so if you still want me to replace it.

Reply:
I saw that you used `for...of` before,  just wanted to keep it consistent

Reply:
I see.
the reason was I needed both sides (keys and values) of the object, and using the forEach method would decrease the code readability.

btw, thanx for the merge buddy